src/tool/cpp/cppwinrt/main.cpp

```diff
@@ -13,37 +14,105 @@ namespace xlang
 {
     settings_type settings;
 
-    struct usage_exception {};
+    struct usage_exception { bool details{}; };
 
-    static void process_args(int const argc, char** argv)
+    static std::vector<cmd::option> options
+    {
+        { "input", 0, cmd::option::no_max, "<spec>", "Windows metadata to include in projection" },
+        { "reference", 0, cmd::option::no_max, "<spec>", "Windows metadata to reference from projection" },
+        { "output", 0, 1, "<path>", "Location of generated projection and component templates" },
+        { "component", 0, 1, "[<path>]", "Generate component templates, and optional implementation" },
+        { "name", 0, 1, "<name>", "Specify explicit name for component files" },
+        { "verbose", 0, 0, {}, "Show detailed progress information" },
+        { "overwrite", 0, 0, {}, "Overwrite generated component files" },
+        { "prefix", 0, cmd::option::no_max, {}, "Use dotted namespace convention for component files (defaults to folders)" },
+        { "pch", 0, 1, "<name>", "Specify name of precompiled header file (defaults to pch.h)" },
+        { "include", 0, cmd::option::no_max, "<prefix>", "One or more prefixes to include in input" },
+        { "exclude", 0, cmd::option::no_max, "<prefix>", "One or more prefixes to exclude from input" },
+        { "base", 0, 0, {}, "Generate base.h unconditionally" },
+        { "opt", 0, 0, {}, "Generate component projection with unified construction support" },
+        { "help", 0, cmd::option::no_max, {}, "Show detailed help with examples" },
+        { "root", 0, 1 }, // "Root folder name for projection headers (defaults to winrt)
+        { "filter" }, // One or more prefixes to include in input (same as -include)
+        { "lib", 0, 1 }, // Specify library prefix (defaults to winrt)
+        { "license", 0, 0 }, // Generate license comment
+        { "brackets", 0, 0 }, // Use angle brackets for #includes (defaults to quotes)
+    };
+
+    static void printUsage(writer& w, bool details)
```

Comment:
The time you're spending with Java is rubbing off on you. :wink:

`print_usage`

Reply:
yes, it is!